ext/src/Cassandra/Types/Float.c

```diff
@@ -0,0 +1,446 @@
+  cassandra_float* result =
+      (cassandra_float*) zend_object_store_get_object(return_value TSRMLS_CC);
+  result->value = fabsf(self->value);
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::neg() */
+PHP_METHOD(Float, neg)
+{
+  cassandra_float* self =
+      (cassandra_float*) zend_object_store_get_object(getThis() TSRMLS_CC);
+  object_init_ex(return_value, cassandra_float_ce);
+  cassandra_float* result =
+      (cassandra_float*) zend_object_store_get_object(return_value TSRMLS_CC);
+  result->value = -self->value;
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::sqrt() */
+PHP_METHOD(Float, sqrt)
+{
+  cassandra_float* self =
+      (cassandra_float*) zend_object_store_get_object(getThis() TSRMLS_CC);
+
+  if (self->value < 0) {
+    zend_throw_exception_ex(cassandra_range_exception_ce, 0 TSRMLS_CC,
+                            "The operation would result in a complex number");
+  }
+
+  object_init_ex(return_value, cassandra_float_ce);
+  cassandra_float* result =
+      (cassandra_float*) zend_object_store_get_object(return_value TSRMLS_CC);
+  result->value = sqrtf(self->value);
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::toInt() */
+PHP_METHOD(Float, toInt)
+{
+  cassandra_float* self =
+      (cassandra_float*) zend_object_store_get_object(getThis() TSRMLS_CC);
+
+  RETURN_LONG((long) self->value);
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::toDouble() */
+PHP_METHOD(Float, toDouble)
+{
+  cassandra_float* self =
+      (cassandra_float*) zend_object_store_get_object(getThis() TSRMLS_CC);
+
+  RETURN_DOUBLE((double) self->value);
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::minValue() */
+PHP_METHOD(Float, minValue)
+{
+  object_init_ex(return_value, cassandra_float_ce);
+  cassandra_float* flt =
+          (cassandra_float*) zend_object_store_get_object(return_value TSRMLS_CC);
+  flt->value = LONG_LONG_MIN;
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::minValue() */
+PHP_METHOD(Float, maxValue)
+{
+  object_init_ex(return_value, cassandra_float_ce);
+  cassandra_float* flt =
+          (cassandra_float*) zend_object_store_get_object(return_value TSRMLS_CC);
+  flt->value = LONG_LONG_MAX;
+}
+/* }}} */
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo__construct, 0, ZEND_RETURN_VALUE, 1)
+  ZEND_ARG_INFO(0, value)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_none, 0, ZEND_RETURN_VALUE, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_num, 0, ZEND_RETURN_VALUE, 1)
+  ZEND_ARG_INFO(0, num)
+ZEND_END_ARG_INFO()
+
+static zend_function_entry cassandra_float_methods[] = {
+  PHP_ME(Float, __construct, arginfo__construct, ZEND_ACC_CTOR|ZEND_ACC_PUBLIC)
+  PHP_ME(Float, __toString, arginfo_none, ZEND_ACC_PUBLIC)
+  PHP_ME(Float, value, arginfo_none, ZEND_ACC_PUBLIC)
+  PHP_ME(Float, isInfinite, arginfo_none, ZEND_ACC_PUBLIC)
+  PHP_ME(Float, isFinite, arginfo_none, ZEND_ACC_PUBLIC)
+  PHP_ME(Float, isNaN, arginfo_none, ZEND_ACC_PUBLIC)
+  PHP_ME(Float, add, arginfo_num, ZEND_ACC_PUBLIC)
```

Comment:
I thought about that, but I was thinking these could also accept php long/double/string?